### PR TITLE
Add container registry checker

### PR DESCRIPTION
**Pull request by davidstacyibm**
this adds a check for the container registry - if its not ready - patch it for empty dir, which fixes on OCPv

### CODE REVIEW

**Comment by davidstacyibm** [comment]
we could probably expand more logic

found out on OCPv that by default the openshift container registry operator is installed but not configured with any storage. 

This change checks for readiness and patches the storage to emptyDir, and sets state to manage so that a container registry deployment and pod is created.

emptyDir was found to work best and doesnt use any back end storage, we do not anticipate ever needing more than one replica for this purpose, but if we did would probably use a PVC.

**Comment by rampadc** [comment]
@davidstacyibm While the script in your task works perfectly fine locally, I could not get $status to set correctly in cluster. Most likely permission issues with the task. However, if you were to use the existing `ibm-pak-0.3` that comes with the cluster, works fine, nothing needs to be changed.

Please see an example pipeline below. I also added in a while loop to wait for the container registry to become ready.

```yaml
apiVersion: tekton.dev/v1
kind: Pipeline
metadata:
  name: check-container-registry
  namespace: default
spec:
  tasks:
    - name: check-container-registry
      params:
        - name: SCRIPT
          value: |
            status=$(oc get config -o jsonpath='{.items[0].status.conditions[?(@.type=="Available")].reason}')

            TIMEOUT_SECONDS=300
            while [[ "$status" != "Ready" ]] && [[ $SECONDS -lt $TIMEOUT_SECONDS ]]; do
              echo "OpenShift Container Registry is not ready. Current status: $status"

              # Apply the patch commands
              oc patch config.imageregistry.operator.openshift.io/cluster --type=merge -p '{"spec":{"storage":{"emptyDir":{}}}}'
              oc patch config.imageregistry.operator.openshift.io/cluster --type=merge -p '{"spec":{"managementState":"Managed"}}'

              # Check status after a short delay to avoid overwhelming the server
              sleep 10

              # Update status (replace with your actual way to get the status)
              status=$(oc get config -o jsonpath='{.items[0].status.conditions[?(@.type=="Available")].reason}')
            done

            if [[ "$status" == "Ready" ]]; then
              echo "OpenShift Container Registry is ready"
              exit 0
            else
              echo "OpenShift Container Registry failed to become ready after $TIMEOUT_SECONDS seconds." >&2
              exit 1
            fi
      taskRef:
        kind: Task
        name: ibm-pak-0.3
```

The above pipeline has the following log output

```
OpenShift Container Registry is not ready. Current status: Removed
config.imageregistry.operator.openshift.io/cluster patched
config.imageregistry.operator.openshift.io/cluster patched
OpenShift Container Registry is not ready. Current status: NoReplicasAvailable
config.imageregistry.operator.openshift.io/cluster patched (no change)
config.imageregistry.operator.openshift.io/cluster patched (no change)
OpenShift Container Registry is not ready. Current status: NoReplicasAvailable
config.imageregistry.operator.openshift.io/cluster patched (no change)
config.imageregistry.operator.openshift.io/cluster patched (no change)
OpenShift Container Registry is ready
```

File: 9.0.x/pipeline.yaml
Comment (rampadc) [inline 9.0.x/pipeline.yaml:1]: Task doesn't assign `status` correctly

**Review by rampadc — CHANGES_REQUESTED** [review]
Please see https://github.com/cloud-native-toolkit/deployer-mas-cli/pull/27#issuecomment-2552799105

**Pull request merged**